addons/account/wizard/account_payment_register.py

```diff
@@ -203,6 +203,9 @@ def _get_line_batch_key(self, line):
         :return: A python dictionary.
         '''
         move = line.move_id
+        early_pay = 0.0
+        if move.is_eligible_for_early_discount(self.payment_date):
+            early_pay = move.early_pay_amount_after_discount
```

Comment:
it's a very bad idea to group on a monetary value :p
Instead, group on the payment term is the invoice is eligible for epd